src/g_doom/a_doomweaps.cpp

```diff
@@ -619,11 +620,15 @@ DEFINE_ACTION_FUNCTION_PARAMS(AActor, A_BFGSpray)
 
 			if (spray != NULL)
 			{
-				if (spray->flags6 & MF6_MTHRUSPECIES && self->target->GetSpecies() == linetarget->GetSpecies())
+				if ((spray->flags6 & MF6_MTHRUSPECIES && self->target->GetSpecies() == linetarget->GetSpecies()) ||
+					(spray->flags6 & MF6_THRUSPECIES && spray->GetSpecies() == linetarget->GetSpecies()))
 				{
 					spray->Destroy(); // [MC] Remove it because technically, the spray isn't trying to "hit" them.
 					continue;
 				}
+				if (spray->flags7 & MF7_HITTARGET)	spray->target = linetarget; //Overridden by PUFFGETSOWNER.
+				if (spray->flags7 & MF7_HITMASTER)	spray->master = linetarget;
+				if (spray->flags7 & MF7_HITTRACER)	spray->tracer = linetarget;

```

Comment:
If anyone is wondering, both linetarget and spray are already being checked for non-null.